### ISSUE TITLE
Line Chart grouped by week of year shows 1 more week

### ISSUE THREAD

**Issue by elton182** [metabase/metabase#8969]
- Your browser and the version:  Chrome 70.0.3538.67
- Your operating system: Korora 26, and Windows 10
- Your databases: MySQL
- Metabase version: v0.31.0
- Metabase hosting environment: Ubuntu server 17.10
- Metabase internal database: H2 (default)

We currently are on week 47 of 2018, but metabase show one more week on the chart.

![image](https://user-images.githubusercontent.com/8716770/48714700-63972000-ebfa-11e8-98ab-dcc972cecb97.png)

When I drill down to the results it shows no data.

![image](https://user-images.githubusercontent.com/8716770/48714790-9ccf9000-ebfa-11e8-8ebf-fb98869f74ce.png)

**Comment by elton182** [comment]
This also happens at the beginning of the year, where it doesn't show the first week

![image](https://user-images.githubusercontent.com/8716770/51200547-f8f41f00-18e0-11e9-8d09-6a8636a6a466.png)

**Comment by flamber** [comment]
@elton182 
I'm not seeing this on 0.31.2 with SQL Server 2008.
Have you checked your timezones - check MySQL (or server of MySQL) and MySQL database/column and Metabase>Settings>Admin>General (or server of Metabase).
Have you seen the troubleshoot guide for timezones?
https://www.metabase.com/docs/latest/troubleshooting-guide/timezones.html

**Comment by felipebueno** [comment]
Hello, @flamber!

I work with @elton182 and I checked all three relevant timezone settings (OS, MySQL and Metabase) and they are all set properly. We also tried setting Report Timezone to Database Default but that didn't work either =/

Prints for reference:
![timezones](https://user-images.githubusercontent.com/241271/51679393-0175e500-1fc6-11e9-961b-c6f23de9c553.png)

![timezones_metabase](https://user-images.githubusercontent.com/241271/51679405-089cf300-1fc6-11e9-937c-19e2e2b3946e.png)

**Comment by flamber** [comment]
@elton182 

Okay, so that seems like Metabase is doing something it shouldn't.

It's setting `mode` to `6`, which is Sunday - most of the world uses Monday as the first day of the week. This shouldn't be defined at all - then it would use the variable `default_week_format`. Another option would be to have a setting in Metabase (just like with other timezone/formatting stuff).
https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_week

Then the `+ 1` is really strange. It shouldn't do that, no matter if `mode` is defined or not.

I can see where the `mode` stuff is coming from, but I have no idea where the `+ 1` is hiding:
https://github.com/metabase/metabase/blob/master/src/metabase/driver/mysql.clj#L211-L213

Just did a couple of tests and seems like only MySQL/MariaDB is behaving this way.

SQL2008 (production data):
```
SELECT datepart(iso_week, "dbo"."Facturas"."fecha") AS "fecha", sum("dbo"."Facturas"."total") AS "sum"
```
H2 (sample data):
```
SELECT week("PUBLIC"."ORDERS"."CREATED_AT") AS "CREATED_AT", sum("PUBLIC"."ORDERS"."TOTAL") AS "sum"
```
MariaDB 10.3 (sample data):
```
SELECT (week(`ORDERS`.`CREATED_AT`, 6) + 1) AS `CREATED_AT`, sum(`ORDERS`.`TOTAL`) AS `sum`
```

**Comment by flamber** [comment]
This was fixed in 0.37.0, when "First day of the week" setting was introduced.
